operator/objectify.go

```diff
@@ -0,0 +1,95 @@
+package operator
+
+import (
+	"encoding/json"
+	"errors"
+
+	_ "github.com/araddon/qlbridge/qlbdriver"
+	"github.com/liminaab/filtrify/types"
+)
+
+type ObjectifyOperator struct {
+}
+
+type ObjectifyConfiguration struct {
+	Fields          []string `json:"fields"`
+	TargetFieldName string   `json:"targetFieldName"`
+}
+
+func (t *ObjectifyOperator) Transform(dataset *types.DataSet, config string, _ map[string]*types.DataSet) (*types.DataSet, error) {
+
+	typedConfig, confError := t.buildConfiguration(config)
+	if confError != nil {
+		return nil, confError
+	}
+
+	newDataset := types.DataSet{
+		Rows: make([]*types.DataRow, len(dataset.Rows)),
+	}
+
+	for i, row := range dataset.Rows {
+		newRow := types.DataRow{
+			Columns: make([]*types.DataColumn, 0),
+		}
+		objectColumnMap := make(map[string]interface{})
+		for _, col := range row.Columns {
+			shouldBeRemoved := false
+			for _, toRemove := range typedConfig.Fields {
+				if col.ColumnName == toRemove {
+					shouldBeRemoved = true
+					break
+				}
+			}
```

Comment:
Maybe make typedConfig.Fields to a map, since the datasets can be pretty large and wide

Reply:
this is the list of columns to be embedded in an object. so not the real data itself

Reply:
Yes, but it will be a third level of looping, so probably worth to optimize when its such a small investment to make a map with the columns instead